normalized-cache-incubating/src/commonMain/kotlin/com/apollographql/cache/normalized/ApolloStore.kt

```diff
@@ -56,20 +57,24 @@ interface ApolloStore {
   /**
    * Reads an operation from the store.
    *
-   * This is a synchronous operation that might block if the underlying cache is doing IO.
+   * When [returnPartialResponses] is `false`, in case of missing data the returned [ApolloResponse.data] is `null` and
+   * [ApolloResponse.exception] is a [com.apollographql.apollo.exception.CacheMissException].
    *
-   * @param operation the operation to read
+   * When [returnPartialResponses] is `true`, the returned [ApolloResponse.data] has `null` values for any missing fields if their
+   * type is nullable, bubbling up to their parent otherwise. Missing fields have a corresponding [com.apollographql.apollo.api.Error]
+   * in [ApolloResponse.errors].
    *
-   * @throws [com.apollographql.apollo.exception.CacheMissException] on cache miss
-   * @throws [com.apollographql.apollo.exception.ApolloException] on other cache read errors
+   * This is a synchronous operation that might block if the underlying cache is doing IO.
    *
-   * @return the operation data with optional headers from the [NormalizedCache]
+   * @param operation the operation to read
+   * @param returnPartialResponses whether to return partial responses
    */
   fun <D : Operation.Data> readOperation(
       operation: Operation<D>,
       customScalarAdapters: CustomScalarAdapters = CustomScalarAdapters.Empty,
       cacheHeaders: CacheHeaders = CacheHeaders.NONE,
-  ): ReadResult<D>
+      returnPartialResponses: Boolean = false,
```

Comment:
Do we need a flag vs just always returning partial responses? 

If you have a partial response, falling back to the throwing case is mostly 

```kotlin
if (response.errors.isNotEmpty()) {
  throw CacheMissException()
}
```

I would get the flag for backward compat but given this is a new artifact, we could do things partial from day 1?

Reply:
> Making this the default is a bigger change for people migrating from the main artifact - which can be OK if it's well documented/called out.

I'd vote for doing partial by default. It makes the API a lot more consistent. 

> Another question is what this does to the CacheFirst policy. 

I'd vote for CacheFirst requires a complete response. Any error triggers a network round trip.